aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2301,9 +2301,12 @@ Tensor &frobenius_norm_out(const Tensor& self,
     maybe_wrap_dims(dim_, self.dim());
     TORCH_CHECK(dim_[0] != dim_[1], "Expected dims to be different, got ", dim, " instead");
     if (self.is_complex()){
-      result_ = at::sqrt(at::sum(at::real(self.conj() * self), dim_, keepdim));
+      // Dont call sqrt if sum() is zero since it will produce nan with autograd.
+      auto sum = at::sum(at::real(self.conj() * self), dim_, keepdim);
```

Comment:
@anjali411 if we add `at::real` for real tensors we could merge both branches of the computation.

Reply:
yeah sounds good! I forget if there's an outstanding PR for it. If not, I can create one!